src/spikeinterface/core/tests/test_template_class.py

```diff
@@ -94,11 +96,8 @@ def test_get_sparse_templates(template_type):
             template.sparsity.max_num_active_channels,
         )
         assert template.are_templates_sparse()
-    elif template_type == "sparse_with_dense_templates":
-        sparse_templates = template.get_sparse_templates()
-        assert sparse_templates.shape == (
-            template.num_units,
-            template.num_samples,
-            template.sparsity.max_num_active_channels,
-        )
-        assert not template.are_templates_sparse()
+
+
+def test_initialization_fail_with_dense_templates():
+    with pytest.raises(ValueError, match="Sparsity mask passed but the templates are not sparse"):
+        template = generate_test_template(template_type="sparse_with_dense_templates")
```

Comment:
@h-mayorquin in this line, would it make sense to have a `Templates.make_sparse(sparsity)` method which returns a `Templates` object with sparse templates using the `sparsity`? (not only the template_array)

Reply:
Seems like an overkill to me. The only thing that is different is the `template_array`. 

If you want to make it simply and unintrusive for users we shouid sparsify ourselves at `__init__` instead of throwing an error. 
If you want to make it simpler but want to let them know we should have a method to sparsify the template array and tell them to use that.

Reply:
got it. So the current way to do that would be:
```
templates_dense_array = ...
sparsity = ...

templates_sparse_array = sparsity.sparsify_waveforms(templates_dense)

templates_sparse = Templated(templates_sparse_array, sparsity.mask)
```

Indeed, looks simple enough  :)